test/e2e/operator_test.go

```diff
@@ -56,6 +64,232 @@ func TestDefaultTunedExists(t *testing.T) {
 		return true, nil
 	})
 	if err != nil {
-		t.Errorf("Failed to get default tuned: %v", err)
+		t.Fatalf("Failed to get default tuned: %v", err)
 	}
 }
+
+// Test the basic functionality of NTO and its operands.  The default sysctl(s)
+// need(s) to be set across the nodes.
+func TestWorkerNodeSysctl(t *testing.T) {
+	sysctlVar := "net.ipv4.neigh.default.gc_thresh1"
+	client := ntoclient.GetClient()
+	if client == nil {
+		t.Fatal("Failed to get kube client.")
+	}
+
+	nodes, err := getNodesByRole(&client, "worker")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	node := nodes[0]
+	t.Logf("Getting a tuned pod running on node %s", node.Name)
+	pod, err := getTunedForNode(&client, &node)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Ensuring the default worker node profile was set")
+	err = ensureSysctl(sysctlVar, pod, "8192")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// Test the application (and rollback) of a custom profile via pod labelling.
+func TestCustomProfileIngress(t *testing.T) {
+	const (
+		profileIngress  = "../../examples/ingress.yaml"
+		podLabelIngress = "tuned.openshift.io/ingress"
+		sysctlVar       = "net.ipv4.tcp_tw_reuse"
+	)
+
+	client := ntoclient.GetClient()
+	if client == nil {
+		t.Fatal("Failed to get kube client.")
+	}
+
+	t.Logf("Getting a list of worker nodes")
+	nodes, err := getNodesByRole(&client, "worker")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	node := nodes[0]
+	t.Logf("Getting a tuned pod running on node %s", node.Name)
+	pod, err := getTunedForNode(&client, &node)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Getting the current value of %s in pod %s", sysctlVar, pod.Name)
+	valOrig, err := getSysctl(sysctlVar, pod)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Logf("Labelling pod %s with label %s", pod.Name, podLabelIngress)
+	out, err := exec.Command("oc", "label", "pod", "--overwrite", "-n", ntoconfig.OperatorNamespace(), pod.Name, podLabelIngress+"=").CombinedOutput()
+	if err != nil {
+		t.Fatal(fmt.Errorf("%v", string(out)))
+	}
+
+	t.Logf("Applying the custom ingress profile from %s", profileIngress)
+	out, err = exec.Command("oc", "apply", "-n", ntoconfig.OperatorNamespace(), "-f", profileIngress).CombinedOutput()
```

Comment:
Is this step creating the custom profile so that NTO can pick it up and apply it based on the previous label you set? Or how is this being applied?

Reply:
That's exactly the correct understadning.